test/new-e2e/tests/cspm/cspm_test.go

```diff
@@ -0,0 +1,298 @@
+)
+
+type cspmTestSuite struct {
+	e2e.BaseSuite[environments.Kubernetes]
+}
+
+type findings = map[string][]map[string]string
+
+var expectedFindingsMasterEtcdNode = findings{
+	"cis-kubernetes-1.5.1-1.1.12": []map[string]string{
+		{
+			"result": "failed",
+		},
+	},
+	"cis-kubernetes-1.5.1-1.2.16": []map[string]string{
+		{
+			"result": "failed",
+		},
+	},
+	"cis-kubernetes-1.5.1-1.2.19": []map[string]string{
+		{
+			"result": "failed",
+		},
+	},
+	"cis-kubernetes-1.5.1-1.2.21": []map[string]string{
+		{
+			"result": "failed",
+		},
+	},
+	"cis-kubernetes-1.5.1-1.2.22": []map[string]string{
+		{
+			"result": "failed",
+		},
+	},
+	"cis-kubernetes-1.5.1-1.2.23": []map[string]string{
+		{
+			"result": "failed",
+		},
+	},
+	"cis-kubernetes-1.5.1-1.2.24": []map[string]string{
+		{
+			"result": "failed",
+		},
+	},
+	"cis-kubernetes-1.5.1-1.2.25": []map[string]string{
+		{
+			"result": "failed",
+		},
+	},
+	"cis-kubernetes-1.5.1-1.2.26": []map[string]string{
+		{
+			"result": "failed",
+		},
+	},
+	"cis-kubernetes-1.5.1-1.2.33": []map[string]string{
+		{
+			"result": "failed",
+		},
+	},
+	"cis-kubernetes-1.5.1-1.2.6": []map[string]string{
+		{
+			"result": "failed",
+		},
+	},
+	"cis-kubernetes-1.5.1-1.3.2": []map[string]string{
+		{
+			"result": "failed",
+		},
+	},
+	"cis-kubernetes-1.5.1-1.3.3": []map[string]string{
+		{
+			"result": "passed",
+		},
+	},
+	"cis-kubernetes-1.5.1-1.3.4": []map[string]string{
+		{
+			"result": "passed",
+		},
+	},
+	"cis-kubernetes-1.5.1-1.3.5": []map[string]string{
+		{
+			"result": "passed",
+		},
+	},
+	"cis-kubernetes-1.5.1-1.3.6": []map[string]string{
+		{
+			"result": "failed",
+		},
+	},
+	"cis-kubernetes-1.5.1-1.3.7": []map[string]string{
+		{
+			"result": "passed",
+		},
+	},
+	"cis-kubernetes-1.5.1-1.4.1": []map[string]string{
+		{
+			"result": "failed",
+		},
+	},
+	"cis-kubernetes-1.5.1-3.2.1": []map[string]string{
+		{
+			"result": "failed",
+		},
+	},
+}
+var expectedFindingsWorkerNode = findings{
+	"cis-kubernetes-1.5.1-4.2.1": []map[string]string{
+		{
+			"result": "failed",
+		},
+	},
+	"cis-kubernetes-1.5.1-4.2.3": []map[string]string{
+		{
+			"result": "failed",
+		},
+	},
+	"cis-kubernetes-1.5.1-4.2.4": []map[string]string{
+		{
+			"result": "failed",
+		},
+	},
+	"cis-kubernetes-1.5.1-4.2.5": []map[string]string{
+		{
+			"result": "failed",
+		},
+	},
+	"cis-kubernetes-1.5.1-4.2.6": []map[string]string{
+		{
+			"result": "failed",
+		},
+	},
+	"cis-kubernetes-1.5.1-4.2.10": []map[string]string{
+		{
+			"result": "failed",
+		},
+	},
+	"cis-kubernetes-1.5.1-4.2.12": []map[string]string{
+		{
+			"result": "failed",
+		},
+	},
+}
+
+func TestCSPM(t *testing.T) {
+	values := `
+datadog:
+  securityAgent:
+    compliance:
+      enabled: true`
+
+	e2e.Run(t, &cspmTestSuite{}, e2e.WithProvisioner(localkubernetes.Provisioner(localkubernetes.WithAgentOptions(kubernetesagentparams.WithHelmValues(values), kubernetesagentparams.WithoutDualShipping()))))
```

Comment:
<!-- Questions are appropriate if you have a potential concern but are not quite sure if it's relevant or not. Asking the author for clarification or investigation can lead to a quick resolution. -->
**❓ question**
Should this use the `awskind` provisioner?

Reply:
Oh yeah, used the local provisioner to write the test but still need to change that